var/spack/repos/builtin/packages/curl/package.py

```diff
@@ -114,9 +114,10 @@ class Curl(NMakePackage, AutotoolsPackage):
     depends_on("gnutls", when="tls=gnutls")
 
     with when("tls=mbedtls"):
-        depends_on("mbedtls@:2 +pic", when="@:7.78")
-        depends_on("mbedtls@2: +pic", when="@7.79:")
-        depends_on("mbedtls@3.6.0: +pic", when="@8.8.0:")
+        depends_on("mbedtls +pic")
+        depends_on("mbedtls@:2", when="@:7.78")
+        depends_on("mbedtls@:3.5", when="@:8.7")
+        depends_on("mbedtls@2:", when="@7.79:")
```

Comment:
```suggestion
        depends_on("mbedtls@2:", when="@7.79:")
        depends_on("mbedtls@3.2:", when="@8.8")  # https://github.com/curl/curl/issues/13748
```
With this the gitlab CI stack should pick 8.7.1 again, but in #46960 there will still be a way out by picking 8.10.1.